src/components/PaymentPlanHeadPanel.js

```diff
@@ -166,7 +247,7 @@ class PaymentPlanHeadPanel extends FormPanel {
                                 withNull={true}
                                 label={formatMessage(intl, "paymentPlan", "benefitPlan")}
                                 required
-                                value={!!paymentPlan.product ? paymentPlan.product : null}
+                                value={paymentPlan.benefitPlan !== undefined && paymentPlan.benefitPlan !== null ? (isEmptyObject(paymentPlan.benefitPlan) ? null : paymentPlan.benefitPlan) : null}
```

Comment:
It might be good idea to use _isNil_ from lodash here to check if benefitPlan is undefined/null.
https://lodash.com/docs/4.17.15#isNil

Reply:
I checked. Unfortunately this picker works badly after your proposition of change. Maybe we can think about this later when we have more time.